polkadot/node/collation-generation/src/lib.rs

```diff
@@ -178,175 +169,136 @@ impl CollationGenerationSubsystem {
 			},
 		}
 	}
-}
 
-#[overseer::subsystem(CollationGeneration, error=SubsystemError, prefix=self::overseer)]
-impl<Context> CollationGenerationSubsystem {
-	fn start(self, ctx: Context) -> SpawnedSubsystem {
-		let future = async move {
-			self.run(ctx).await;
-			Ok(())
-		}
-		.boxed();
+	async fn handle_submit_collation<Context>(
+		&mut self,
+		params: SubmitCollationParams,
+		ctx: &mut Context,
+	) -> Result<()> {
+		let Some(config) = &self.config else {
+			return Err(Error::SubmittedBeforeInit);
+		};
+		let _timer = self.metrics.time_submit_collation();
 
-		SpawnedSubsystem { name: "collation-generation-subsystem", future }
-	}
-}
+		let SubmitCollationParams {
+			relay_parent,
+			collation,
+			parent_head,
+			validation_code_hash,
+			result_sender,
+			core_index,
+		} = params;
 
-#[overseer::contextbounds(CollationGeneration, prefix = self::overseer)]
-async fn handle_new_activations<Context>(
-	config: Arc<CollationGenerationConfig>,
-	activated: impl IntoIterator<Item = Hash>,
-	ctx: &mut Context,
-	metrics: Metrics,
-) -> crate::error::Result<()> {
-	// follow the procedure from the guide:
-	// https://paritytech.github.io/polkadot-sdk/book/node/collators/collation-generation.html
-
-	// If there is no collation function provided, bail out early.
-	// Important: Lookahead collator and slot based collator do not use `CollatorFn`.
-	if config.collator.is_none() {
-		return Ok(())
-	}
+		// We need to swap the parent-head data, but all other fields here will be correct.
```

Comment:
nit: This comment is a bit confusing, you swap it LoC later on. Maybe move it there.